src/doc/grid.cpp

```diff
@@ -177,4 +179,60 @@ std::vector<gfx::Point> Grid::tilesInCanvasRegion(const gfx::Region& rgn) const
   return result;
 }
 
+Grid::IsometricGuide::IsometricGuide(const gfx::Size& sz)
+{
+  evenWidth = sz.w % 2 == 0;
+  evenHeight = sz.h % 2 == 0;
+  evenHalfWidth = ((sz.w / 2) % 2) == 0;
+  evenHalfHeight = ((sz.h / 2) % 2) == 0;
+  squareRatio = ABS(sz.w - sz.h) <= 1;
+  oddSize = !evenWidth && evenHalfWidth && !evenHeight && evenHalfHeight;
+
+  // TODO: add 'share edges' checkbox to UI.
+  // For testing the option, set this 'false' to 'true'
+  shareEdges = !(false && !squareRatio && evenHeight);
```

Comment:
I'm thinking if the shareEdges could be an argument of this function, or another type of grid `IsometricSharedEdges` or something like that (and passing the grid type to `IsometricGuide` ctor).